File: tags/com.sos.scheduler.engine.pom-1.3.11-beta-1/playground/zschimmer/src/main/scala/com/sos/scheduler/engine/playground/zschimmer/PlugInFactory.scala
package com.sos.scheduler.engine.playground.zschimmer

import com.sos.scheduler.engine.kernel.Scheduler
import com.sos.scheduler.engine.kernel.plugin.PlugIn
import com.sos.scheduler.engine.kernel.plugin.{PlugInFactory => JavaPlugInFactory}
import org.w3c.dom.Element
import scala.xml.Elem


trait PlugInFactory extends JavaPlugInFactory {
    def newInstance(scheduler: Scheduler, configurationOption: Option[Elem]): PlugIn

    override def newInstance(scheduler: Scheduler, configurationOrNull: Element) = {
        val elem = Option(configurationOrNull) map { XMLs.fromJavaDom(_).asInstanceOf[xml.Elem] }
        newInstance(scheduler, elem)
    }
}

File: opensource/kernel/src/main/java/com/sos/scheduler/engine/kernel/plugin/PluginFactory.java
package com.sos.scheduler.engine.kernel.plugin;

import com.sos.scheduler.engine.kernel.Scheduler;

import org.w3c.dom.Element;

/** Ein Plugin muss eine statische Methode implementieren, die Objekt dieser Klasse liefert.
 * Die Methode wird über Reflection aufgerufen.
 * 
 * public static PluginFactory newFactory();
*/
public interface PluginFactory {
    Plugin newInstance(Scheduler scheduler, Element configurationOrNull);
}
